src/services/validation.js

```diff
@@ -1,36 +1,41 @@
 import firebaseAPI from "@/services/API/checkUser.js";
 import { sendPlio } from "@/services/sendPlio";
+import {sendSQSMessage} from "@/services/API/sqs";
 
-//this function is invoked only for the check of SRN's.
-export async function validateSRN(userID, validateCount, isSingleEntryOnly, redirectID, doesUserExist){
+// this function is invoked only for the check of SRN's.
+// validateCount = 0, when the user gets validated on the first try
+// validateCount = 1, when the user fails the first validation
 
+export async function validateSRN(userID, validateCount, isSingleEntryOnly, redirectID, doesUserExist, purpose, purposeParams, redirectTo){
+    var authType="SRN"
```

Comment:
This method seems to be doing two things: validating and sending the SQS message. Split into two separate methods. The send sqs message logic should the responsibility of the caller

Reply:
It doesn't necessarily contain the SQS logic. It only calls the send SQS message function. That function which is in file `sqs.js` contains the logic.

Reply:
Yes, this method shouldn't be responsible for that -- that's the point. This method should only validate the SRN and return true or false. The `sqs.js` file should contain ALL the logic for interacting with sqs.